src/main/java/com/minecolonies/core/client/gui/citizen/MainWindowCitizen.java

```diff
@@ -103,14 +103,14 @@ public void onButtonClicked(@NotNull final Button button)
             final String label = button.getID().replace(PLUS_PREFIX, "");
             final Skill skill = Skill.valueOf(StringUtils.capitalize(label));
 
-            Network.getNetwork().sendToServer(new AdjustSkillCitizenMessage(colony, citizen, 1, skill));
+            Network.getNetwork().sendToServer(new AdjustSkillCitizenMessage(citizen.getColony(), citizen, 1, skill));
```

Comment:
instead of all of these changes and requiring the citizen view to have the colony view method, you could've just had the parent WindowCitizen building have the cached colony instead